### DISCUSSION TITLE
Alias resolution

### DISCUSSION THREAD

**Discussion by Paulalex85** [intellij-solidity/intellij-solidity#459] [General]
I'm currently implementing the resolution of alias and I'm not sure about where it should resolve.

Firstly, when resolving the alias in the import, like `import {a as A} from "./a.sol";` or `import "./a.sol" as A;`, should I implement a resolution? It's currently not the case. It will resolve to the reference of `a` in the first example and to the file `a.sol` in the second one. 

Secondly, when resolving the alias when it's used in the code, should it resolve to the declaration of the alias or the reference of what the alias refers to ?
Example, in which //x it should resolve ? 
```
pragma solidity ^0.8.26;
    
library a {
      //x
    function doit() internal {
    }
}
File "a.sol"


pragma solidity ^0.8.26;
    
import {a as A} from "./a.sol";
           //x
contract b {
    function test() public {
        A.doit();
      //^
    }
}
File "b.sol"
```

@mbolotov @SerCeMan 
Thank you

**Comment by SerCeMan** [comment]
I think the best approach would be to keep the behaviour consistent with other language plugins to make it less surprising and more intuitive. For import aliases, we could possibly borrow the behaviour from the typescript plugin.

**Reply by Paulalex85** [reply]
It seem that for the first one, it's going to the usages of the alias.
For the second one, it's going the library implementation. 
I'm gonna try to implement this

**Reply by mbolotov** [reply]
Personally, I like the first approach. That being said, I agree that making the consistent behavior may with other language be a better choice (AFAIK, the typescript plugin adhere to the second one.)

**Reply by Paulalex85** [reply]
So for my first point it will be like this 
```
pragma solidity ^0.8.26;
    
import {a as A} from "./a.sol";
           //^ 
contract b {
    function test() public {
        A.doit();
      //x
    }
}
```

and for my second point it's will be like this 

```
pragma solidity ^0.8.26;
    
library a {
      //x
    function doit() internal {
    }
}
File "a.sol"


pragma solidity ^0.8.26;
    
import {a as A} from "./a.sol";

contract b {
    function test() public {
        A.doit();
      //^
    }
}
File "b.sol"
```

Thanks for your quick answers
